### PR TITLE
HDDS-3399. Update JaegerTracing

### PR DESCRIPTION
**Pull request by elek**
## What changes were proposed in this pull request?

We currently use JaegerTracing 0.34.0. The latest is 1.2.0. We are several versions behind and should update. Note this update requires the latest version fo OpenTracing and has several breaking changes.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3399

## How was this patch tested?

```
cd hadoop-ozone/dist/target/ozone-..../compose/ozone
export COMPOSE_FILE=docker-compose.yaml:profiling.yaml:monitoring.yaml
export OZONE_REPLICATION_FACTOR=3
./run -d
```
Executed freon + s3 crete bucket commands

Traces were displayed in jaeger.

### CODE REVIEW

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/tracing/TracingUtil.java [hunk 1]
```diff
@@ -129,8 +132,53 @@ private static SpanContext extractParent(String parent, Tracer tracer) {
...
-          ScmConfigKeys.HDDS_TRACING_ENABLED,
-          ScmConfigKeys.HDDS_TRACING_ENABLED_DEFAULT);
+        ScmConfigKeys.HDDS_TRACING_ENABLED,
+        ScmConfigKeys.HDDS_TRACING_ENABLED_DEFAULT);
+  }
+
+  /**
+   * Execute a new function inside an activated span.
+   */
+  public static <R> R executeInsideNewSpan(String spanName,
```
Comment (adoroszlai) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/tracing/TracingUtil.java:142]: Nit: can we simplify to `executeInNewSpan`?

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/tracing/TracingUtil.java [hunk 2]
```diff
@@ -129,8 +132,53 @@ private static SpanContext extractParent(String parent, Tracer tracer) {
...
+      SupplierWithIOException<R> supplier)
+      throws IOException {
+    Span span = GlobalTracer.get()
+        .buildSpan(spanName).start();
+    try (Scope scope = GlobalTracer.get().activateSpan(span)) {
+      return supplier.get();
+    } catch (Exception ex) {
+      span.setTag("failed", true);
+      throw ex;
+    }
```
[the remark this answers is not in the document]
Comment (elek) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/tracing/TracingUtil.java:152]: Thanks to test it @adoroszlai . I didn't notice any problem with the spans when I tested, but yes, it seems to be missing. Added them.

[the remark this answers is not in the document]
Comment (adoroszlai) [reply hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/tracing/TracingUtil.java:152]: master: s3shell_master
patch: s3shell_patch

**Comment by elek** [comment]
Thanks the review @adoroszlai I am merging it now (rebased checked with CI and got green light).

**Pull request closed without merging**